### PR TITLE
feat: add sharepoint site id option

### PR DESCRIPTION
**Pull request by EmanuelCampos**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 1]
```diff
@@ -138,10 +141,16 @@ def _get_site_id_with_host_name(self, access_token, sharepoint_site_name) -> str
         if hasattr(self, "_site_id_with_host_name"):
             return self._site_id_with_host_name
 
-        site_information_endpoint = f"https://graph.microsoft.com/v1.0/sites"
-
         self._authorization_headers = {"Authorization": f"Bearer {access_token}"}
 
+        if self.sharepoint_site_id:
+            sharepoint_side_id = self.sharepoint_site_id
```
Comment (masci) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:147]: I think you don't need to assign `sharepoint_site_id` (there's a typo BTW), you can just use `self.sharepoint_site_id` at line 149

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 2]
```diff
@@ -546,8 +556,11 @@ def load_data(
         if not sharepoint_folder_id:
             sharepoint_folder_id = self.sharepoint_folder_id
 
+        if not sharepoint_site_id:
```
Comment (masci) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:559]: `load_data` should take a new parameter `sharepoint_site_id` I think?

**Review by masci — CHANGES_REQUESTED** [review]
left a couple of comments

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py (hunk 2 above)
Comment (EmanuelCampos) [reply llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py]: just moved the check to line 560 now, keepin only on the initialization to not pass site_id all the way down

**Review by masci — APPROVED** [review]
LGTM!

**Pull request merged**